### PR TITLE
Reorganize unit test layout

### PR DESCRIPTION
**Pull request by hubertdeng123**
This PR separates out unit tests into a separate folder

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
         uses: actions/checkout@v2
 
       - name: Unit Tests
-        working-directory: install
+        working-directory: _unit-test
         run: find ./ -type f -name "*-test.sh" -exec "./{}" \;
```
Comment (chadwhitacre) [inline .github/workflows/test.yml:26]: Wait so does this work after all?

Comment (hubertdeng123) [reply .github/workflows/test.yml:26]: it runs the tests, but does not alert on failures

Comment (hubertdeng123) [reply .github/workflows/test.yml:26]: so no :(

File: install/error-handling-test.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
```
Comment (chadwhitacre) [inline install/error-handling-test.sh:1]: Does this file belong here?

Comment (hubertdeng123) [reply install/error-handling-test.sh]: yeah good catch

File: _unit-test/_test_setup.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
+set -euo pipefail
```
Comment (chadwhitacre) [inline _unit-test/_test_setup.sh:1]: Should this be removed from `install/`?

Comment (hubertdeng123) [reply _unit-test/_test_setup.sh:1]: all these duplicate files should be removed

**Comment by chadwhitacre** [comment]
Runs clean for me. 👍

Screen Shot 2022-09-29 at 5 11 08 PM

**Review by chadwhitacre — APPROVED** [review]

**Review by chadwhitacre — CHANGES_REQUESTED** [review]
Hrm ... rerun fails though.

```
$ ./unit-test.sh && echo pass || echo fail
Cloning into '/tmp/sentry-self-hosted-test-sandbox.zow'...
Cloning into '/tmp/sentry-self-hosted-test-sandbox.6B0'...
▶ Initializing Docker Compose ...

▶ Ensuring Relay credentials ...
Creating ../relay/config.yml...
relay Pulling 
relay Pulled 
Relay credentials written to ../relay/credentials.json.

▶ Ensuring Relay credentials ...
../relay/config.yml already exists, skipped creation.
../relay/credentials.json already exists, skipped creation.

ensure-relay-credentials-test.sh - Success 👍
Cloning into '/tmp/sentry-self-hosted-test-sandbox.12Q'...
▶ Setting up error handling ...
latest: Pulling from getsentry/sentry-cli
Digest: sha256:03980e3e7526265ec1db9f8e9847b880521057844c4d93c8e56f8872cd51f8d0
Status: Image is up to date for getsentry/sentry-cli:latest
docker.io/getsentry/sentry-cli:latest

Testing initial send_event
Pass.
Testing send_event duplicate
Pass.
Testing cleanup without minimizing downtime
Pass.
Testing cleanup while minimizing downtime
Pass.
error-handling-test.sh - Success 👍
Cloning into '/tmp/sentry-self-hosted-test-sandbox.vYw'...
▶ Setting up GeoIP integration ...
Setting up IP address geolocation ...
Installing (empty) IP address geolocation database ... done.
IP address geolocation is not configured for updates.
See https://develop.sentry.dev/self-hosted/geolocation/ for instructions.
Error setting up IP address geolocation.

▶ Setting up GeoIP integration ...
Setting up IP address geolocation ...
IP address geolocation database already exists.
IP address geolocation is not configured for updates.
See https://develop.sentry.dev/self-hosted/geolocation/ for instructions.
Error setting up IP address geolocation.

geoip-test.sh - Success 👍
fail
$
```

**Comment by chadwhitacre** [comment]
With latest commits:

```
$ ./unit-test.sh
🙈 Running ./_unit-test/create-docker-volumes-test.sh ...
Cloning into '/tmp/sentry-self-hosted-test-sandbox.Gep'...
fail 👎
🙈 Running ./_unit-test/ensure-relay-credentials-test.sh ...
Cloning into '/tmp/sentry-self-hosted-test-sandbox.qO4'...
▶ Initializing Docker Compose ...

▶ Ensuring Relay credentials ...
Creating ../relay/config.yml...
relay Pulling
relay Pulled
Relay credentials written to ../relay/credentials.json.

▶ Ensuring Relay credentials ...
../relay/config.yml already exists, skipped creation.
../relay/credentials.json already exists, skipped creation.

ensure-relay-credentials-test.sh - Success 👍
🙈 Running ./_unit-test/error-handling-test.sh ...
Cloning into '/tmp/sentry-self-hosted-test-sandbox.0Ul'...
▶ Setting up error handling ...
latest: Pulling from getsentry/sentry-cli
Digest: sha256:03980e3e7526265ec1db9f8e9847b880521057844c4d93c8e56f8872cd51f8d0
Status: Image is up to date for getsentry/sentry-cli:latest
docker.io/getsentry/sentry-cli:latest

Testing initial send_event
Pass.
Testing send_event duplicate
Pass.
Testing cleanup without minimizing downtime
Pass.
Testing cleanup while minimizing downtime
Pass.
error-handling-test.sh - Success 👍
🙈 Running ./_unit-test/geoip-test.sh ...
Cloning into '/tmp/sentry-self-hosted-test-sandbox.kME'...
▶ Setting up GeoIP integration ...
Setting up IP address geolocation ...
Installing (empty) IP address geolocation database ... done.
IP address geolocation is not configured for updates.
See https://develop.sentry.dev/self-hosted/geolocation/ for instructions.
Error setting up IP address geolocation.

▶ Setting up GeoIP integration ...
Setting up IP address geolocation ...
IP address geolocation database already exists.
IP address geolocation is not configured for updates.
See https://develop.sentry.dev/self-hosted/geolocation/ for instructions.
Error setting up IP address geolocation.

geoip-test.sh - Success 👍
$
```

File: _unit-test/create-docker-volumes-test.sh [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+sentry-postgres
+sentry-redis
+sentry-symbolicator
+sentry-zookeeper"
+
+before=$(get_volumes)
+
+test "$before" == "" || test "$before" == "$expected_volumes"
+
+source create-docker-volumes.sh
```
Comment (chadwhitacre) [inline _unit-test/create-docker-volumes-test.sh:23]: Are we in the right place for this `source` to work? Do the paths line up? Same in other files ...

Comment (hubertdeng123) [reply _unit-test/create-docker-volumes-test.sh:23]: yep, the `_test_setup.sh` file does a cd into the `install` directory

**Comment by chadwhitacre** [comment]
```
fail 👎
```

My bad for not running `reset.sh` first.

**Review by chadwhitacre — APPROVED** [review]

**Pull request merged**